pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -346,6 +346,12 @@ func (o *operatingSystemConfig) updateHashVersioningSecret(ctx context.Context)
 
 		var pools poolHash
 		for _, worker := range o.values.Workers {
+			// we don't need to track the hash version for in-place updates
+			// as the hash version is always 0
```

Comment:
This comment is confusing. We don't have a "hash version 0". Can you improve this so that it is easier digestable for future readers?

Reply:
hashVersion calculation returns zero for inplace, and it is done this way since it is not dependent on any feature gate. I will reword it for better explantion.